File: testbed_php/test/associations/UnidirectionalOneTest.php
<?php

class UnidirectionalOneTest extends UnitTestCase
{

  public function test_ConstructorManySide()
  {
    $s = new StudentAH(1);
    $m = new MentorAH("a",$s);
    $this->assertEqual("a",$m->getName());
    $this->assertEqual(1,$m->getStudent()->getNumber());
  }
  
  public function test_ConstructorCannotSetNull()
  {
  	$this->expectException(new Exception("Unable to create MentorAH due to aStudent"));
    new MentorAH("a",null);
  }   
  

  public function test_deleteLeavesStudentAlone()
  {
    $m = new MentorAH("a",new StudentAH(1));
    $s = $m->getStudent();
    $p = new ProgramAH();
    $s->setProgram($p);
    $m->delete();
    $this->assertEqual(null,$m->getStudent());
    $this->assertEqual(null,$m->getProgram());
    $this->assertEqual($p,$s->getProgram());
    
  }
  
}

File: testbed_ruby/test/associations/unidirectional_one_test.rb
require 'test/unit'
require './src-gen-umple/student_ah'
require './src-gen-umple/mentor_ah'
require './src-gen-umple/program_ah'

module CruiseAssociations
class UnidirectionalOneTest < Test::Unit::TestCase

  def test_ConstructorManySide
    s = StudentAH.new(1)
    m = MentorAH.new("a",s)
    assert_equal("a",m.get_name)
    assert_equal(1,m.get_student.get_number)
  end
  
  def test_ConstructorCannotSetNull
  	assert_raise RuntimeError do #"Unable to create MentorAH due to aStudent"
      MentorAH.new("a",nil)
    end
  end   
  
  def test_deleteLeavesStudentAlone
    m = MentorAH.new("a",StudentAH.new(1))
    s = m.get_student
    p = ProgramAH.new
    s.set_program(p)
    m.delete
    assert_equal(nil,m.get_student)
    assert_equal(nil,m.get_program)
    assert_equal(p,s.get_program)
  end
end
end
